ethers-core/src/utils/mod.rs

```diff
@@ -158,29 +158,32 @@ where
     T: Into<ParseUnits>,
     K: TryInto<Units, Error = ConversionError>,
 {
-    let units = units.try_into()?;
-    match amount.into() {
+    let units: usize = units.try_into()?.into();
+    let amount = amount.into();
+
+    // 78 overflows U256, 77 overflows I256
+    match amount {
+        // 2**256 ~= 1.16e77
+        ParseUnits::U256(_) if units >= 78 => return Err(ConversionError::ParseOverflow),
```

Comment:
Despite the comments `77` and `78` sound like magic numbers. Should we assign constants with meaningful names?

Reply:
+1